src/core/CustomData.cpp

```diff
@@ -143,14 +144,14 @@ bool CustomData::isEmpty() const
     return m_data.isEmpty();
 }
 
-int CustomData::dataSize()
+int CustomData::dataSize() const
 {
     int size = 0;
 
-    QHashIterator<QString, QString> i(m_data);
+    QMapIterator<QString, QVariant> i(m_data);
     while (i.hasNext()) {
         i.next();
-        size += i.key().toUtf8().size() + i.value().toUtf8().size();
+        size += i.key().toUtf8().size() + i.value().toString().toUtf8().size();
```

Comment:
You may look at this again - `QVariant::toString()` does give the correct size for arbitrary variants afaik. To calculate the correct size, one should cast to the correct type and than calculate the size for the specific type. KeePass2 does implement CustomData with a StringDictionary so I would suggest going back to `QString`